File: pizzeria-meal/src/main/java/pl/pizzeria/order/domain/MealRequest.java
package pl.pizzeria.order.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Positive;
import java.util.List;

@Getter
@AllArgsConstructor
@NoArgsConstructor
public class MealRequest {
    @NotNull
    @Positive
    private Long id;

    @NotNull
    @Positive
    private Integer quantity;

    private List<Long> toppings;
    private Long extrasId;
    private Long baseIngredientId;
}

File: angular-app/src/app/model/meal-request.ts
export class MealRequest {
    id: number;
    quantity: number;
    toppings: number[];
    extrasId: number;
    baseIngredientId: number;

    constructor(id: number, quantity: number, toppings: number[], extrasId: number, baseIngredientId: number) {        
        this.id = id;
        this.quantity = quantity;
        this.toppings = toppings;
        this.extrasId = extrasId;
        this.baseIngredientId = baseIngredientId;
    }
}
